### ISSUE TITLE
Crash in mp1.d_s under certain conditions

### ISSUE THREAD

**Issue by jranalli** [chmarti1/PYroMat#27]
Found a crash within d_s. To see the crash, run the following code in version 2.1.8:

```
import pyromat as pm
import numpy as np

h2o = pm.get('mp.H2O')

s = 8
T = np.linspace(275, 1270, 50)
d = h2o.d_s(T=T, s=s)
```
Basically, it happens because of an indexing bug within mp1.py in the d_s routine when the index I is not uniformly true, leading to different size for ```s_``` on line 3479 and ```I```. The fix is to change the following two lines to just reference ```s_``` instead of ```s_[I]```.

```
3488     s_ = self._s(T=T[I], d=da[I])[0]
3489     I[I] = s_ < s[I]
```

**Comment by chmarti1** [comment]
Confirmed.  The cause is correct.  I'll push a patch soon.

**Comment by chmarti1** [comment]
Test of v2.1.9:
```python
>>> import pyromat as pm
>>> h2o=pm.get('mp.H2O')
>>> g = np.random.default_rng()
>>> T = g.uniform(low=273.17, high=1272, size=10000)
>>> d = g.uniform(low=0., high=1000., size=10000)
>>> s = h2o.s(T=T,d=d)
>>> dd = h2o.d_s(T=T,d=d)
>>> np.max(np.abs(dd-d)/d)
1.4081770015078492e-05
```
Maximum error is less than .0015%.  On my system, convergence takes a little less than a second for the 10,000 element array.

Fixed in patch for v2.1.9.
